tree/treeplayer/src/TTreeReaderArray.cxx

```diff
@@ -460,7 +461,7 @@ void ROOT::Internal::TTreeReaderArrayBase::CreateProxy()
          fProxy = 0;
          return;
       }
-      if (!branchActualType) {
+      if (!branchActualType && !isOpaqueTypedef) {
```

Comment:
Isn't that too coarse?  What about is the type is opaque and the user still pass a different type (eg. a long or a float for Double32_t)?

Reply:
In this case we are left with string comparison I am afraid.

Reply:
not really. See below